code/datums/status_effects/debuffs/debuffs.dm

```diff
@@ -690,6 +690,44 @@
 	desc = "You've been zapped with something and your hands can't stop shaking! You can't seem to hold on to anything."
 	icon_state = "convulsing"
 
+/datum/status_effect/repeatedly_electrocute
+	id = "repeated_electrocution"
+	tick_interval = 8 SECONDS
+	alert_type = /atom/movable/screen/alert/status_effect/repeatedly_electrocute
+	/// How much damage to we do every time?
+	var/damage
+	/// What do we print as our damage source?
+	var/atom/shock_source
+	/// Flags to use for shock details
+	var/shock_flags = SHOCK_NOSTUN
+
+/datum/status_effect/repeatedly_electrocute/on_creation(mob/living/new_owner, shock_source = "something", damage = 30)
+	src.damage = damage
+	src.shock_source = shock_source
+	return ..()
+
+/datum/status_effect/repeatedly_electrocute/on_apply()
+	. = ..()
+	owner.electrocute_act(shock_damage = damage, source = shock_source, flags = shock_flags)
+	RegisterSignal(shock_source, COMSIG_QDELETING, PROC_REF(on_source_deleted))
```

Comment:
(Also this will runtime on the default arguments since you default `shock_source = "something"`)